### PR TITLE
deps: bump celestia-node to v0.11.0-rc15

### PR DESCRIPTION
**Pull request by jcstein**
## Overview

This PR matches the latest celestia-node version. However, am running into issues after [running](https://app.warp.dev/block/mZ8nK5hbfZE4yPJNCcuqvX) the [built](https://app.warp.dev/block/KwbHniCF3V0wN1WNJYXY3E) image.

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

**Comment by jcstein** [comment]
likely due to something in the [diff between rc12 and rc13](https://github.com/celestiaorg/celestia-node/compare/v0.11.0-rc12...v0.11.0-rc13)

**Comment by jcstein** [comment]
may be related https://github.com/celestiaorg/celestia-app/pull/2467/files

**Comment by rootulp** [comment]
Although it looks related https://github.com/celestiaorg/celestia-app/pull/2467 modifies the single-node.sh script that lives inside the celestia-app repo. As far as I can tell, this repo copied a lot of that script's contents and used it in entrypoint.sh so I think we may need a modification in entrypoint.sh to make sure we're deleting and creating the directory `~/.celestia-app` correctly.

**Comment by jcstein** [comment]
thanks for checking this out @rootulp - I can't seem to figure it out! but will keep this pr posted

**Comment by jcstein** [comment]
ah, i see that celestia-app version from v0.11.0-rc12 to rc13 of celestia-node goes from v1.0.0-rc13 to v1.0.0-rc14

**Comment by jcstein** [comment]
this is the same thing i saw. i'll be going over with @smuu, too and will follow up here

**Comment by smuu** [comment]
> > do you know what may be going on here @Bidon15 ?
> 
> I tried debugging and failed miserably for a day and a half. @rootulp we are clearing the `/home/celestia/.celestia-app` directory if it's existing. Otherwise we are just logging that a folder never existed in the first place.
> 
> The issue is that we are missing a binary `/bin/celestia-appd` that is actually existing in that path. This is not happening in rc12 but once you bump to rc13 and main - we see such issues.
> 
> @jcstein, I'll need some more time to figure out why this is happening

I would not say that the error states that it does not find the celestia-app binary as the error states "cannot execute: required file not found".
So I understand it in a way, that the system starts the binary but the binary fails with the error message, that a required file is missing that it needs to start.

**Comment by rootulp** [comment]
FWIW I don't think bumping celestia-app's version is causing this b/c I was able to bump to celestia-app v1.0.0-rc16 in https://github.com/rollkit/local-celestia-devnet/compare/main...rootulp:local-celestia-devnet:rp/app-rc-16 and devnet started fine. I think the issue is originating from the celestia-node bump.

**Comment by jcstein** [comment]
elmo

**Comment by jcstein** [comment]
however, using v1.0.0-rc16 of celestia-app + v0.11.0-rc12 of celestia-node [works just as expected](https://app.warp.dev/block/hfLmABj0L60QD10gLl2cdp)...i believe @rootulp is right here and there is some breaking change between rc12 and rc13 of node

**Comment by rootulp** [comment]
> we suspect it is an issue within celestia-app itself that relies on a specific architecture to perform the commands that are failing

This is correct. I verified that the celestia-appd binary is built for aarch64.
When celestia-node rc-13 is used, the Docker image is using x86_64.
When celestia-node rc-12 is used, the Docker image is aarch64.

**Comment by rootulp** [comment]
I thought there may be a quick fix by explicitly depending on the arm64 version of celestia-node with

```
FROM ghcr.io/celestiaorg/celestia-node:v0.11.0-rc13@sha256:4b5b09baafe1263376d1d1f81a9bc444ad0f3e94b5f8cf19806f5b3303c76b51
```

From: https://github.com/celestiaorg/celestia-node/pkgs/container/celestia-node/126803928?tag=v0.11.0-rc13 but that doesn't resolve. My next idea to resolve is the Dockerfile in this repo can start from a clean slate `FROM docker.io/alpine:3.18.3` and then copy the binaries from both the celestia-app image and celestia-node image but I'm a Docker noob so maybe there's a better way to do this now that the issue has been identified. Any ideas @smuu @Bidon15 ?

**Comment by Bidon15** [comment]
After deep diving into the kernels and different ways of building up an image, I finally got the root cause of all evil

```sh
	/lib/ld-musl-x86_64.so.1 (0xffffba211000)
	libc.musl-x86_64.so.1 => /lib/ld-musl-x86_64.so.1 (0xffffba211000)
Error relocating /bin/celestia: unsupported relocation type 7
Error relocating /bin/celestia: unsupported relocation type 7
```

The musl library was panicking as the reference points were at expected arm64 arc, **_BUT_** it turns out that our celestia-node packages that are platformed as `linux/arm64` turns out to be `x86_64`, which created hell. 

As an interim solution, I propose to use and me with rest of devOps team figure out why GA is making our lives so much harder 

` FROM vietcelestia/celestia-node:v0.11.0-rc13`

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 FROM ghcr.io/celestiaorg/celestia-app:v1.0.0-rc14 AS celestia-app
 
-FROM ghcr.io/celestiaorg/celestia-node:v0.11.0-rc12
+FROM ghcr.io/celestiaorg/celestia-node:v0.11.0-rc13
```
Comment (Bidon15) [inline Dockerfile:3]:
```suggestion
FROM vietcelestia/celestia-node:v0.11.0-rc13
```

Comment (jcstein) [reply Dockerfile]: hm. should this be `ghcr.io/vietcelestia/`

Comment (Bidon15) [reply Dockerfile:3]: nope. I'm using dockerhub 😅

Comment (jcstein) [reply Dockerfile:3]: got it! should we use this or maybe wait until a compatible release is made?

Comment (Bidon15) [reply Dockerfile:3]: let's use this in the meantime and get back to compat release once we fix the pipelines

**Review by Bidon15 — DISMISSED** [review]

**Comment by jcstein** [comment]
hmmm it looks like this doesn't have perms to pull from your vietcelestia image @Bidon15? i don't see it on [this page](https://hub.docker.com/u/vietcelestia), so believe it could be private -- unless it is `node`

https://github.com/rollkit/local-celestia-devnet/actions/runs/6312165992/job/17137652590?pr=63#step:4:154
https://github.com/rollkit/local-celestia-devnet/actions/runs/6312165682/job/17137648172?pr=63#step:4:152

File: entrypoint.sh [hunk 2]
```diff
@@ -7,14 +7,15 @@ APP_PATH="/home/celestia/.celestia-app"
 NODE_PATH="/home/celestia/bridge/"
 
 # Check if the folder exists
+echo "🔍 Checking if the folder $APP_PATH exists..."
```
Comment (appetrosyan) [inline entrypoint.sh:10]: This is an excellent way to ensure that most users whose terminals do not support UTF-8 (and that includes `urxvt`, `gnome-terminal`, `mate-terminal` most editors etc. will see something which potentially makes them ask questions. I'd recommend not including these things if possible.

Comment (jcstein) [inline entrypoint.sh:10]:
```suggestion
echo "Checking if the folder $APP_PATH exists..."
```

File: entrypoint.sh [hunk 3]
```diff
@@ -7,14 +7,15 @@ APP_PATH="/home/celestia/.celestia-app"
...
 if [ -d "$APP_PATH" ]; then
   # If it exists, delete it
-  echo "The folder $APP_PATH exists. Deleting it..."
+  echo "👀 The folder $APP_PATH exists. Deleting it..."
```
Comment (jcstein) [inline entrypoint.sh:13]:
```suggestion
  echo "The folder $APP_PATH exists. Deleting it..."
```

File: entrypoint.sh [hunk 4]
```diff
@@ -7,14 +7,15 @@ APP_PATH="/home/celestia/.celestia-app"
...
   rm -rf "$APP_PATH"
-  echo "Folder deleted."
+  echo "👍 Folder deleted."
```
Comment (jcstein) [inline entrypoint.sh:15]:
```suggestion
  echo "Folder deleted."
```

File: entrypoint.sh [hunk 5]
```diff
@@ -7,14 +7,15 @@ APP_PATH="/home/celestia/.celestia-app"
...
 else
   # If it doesn't exist, print a message
-  echo "The folder $APP_PATH does not exist."
+  echo "👍 The folder $APP_PATH does not exist."
```
Comment (jcstein) [inline entrypoint.sh:18]:
```suggestion
  echo "The folder $APP_PATH does not exist."
```

File: entrypoint.sh (hunk 2 above)
Comment (jcstein) [reply entrypoint.sh]: good call, thank you @appetrosyan. done in https://github.com/rollkit/local-celestia-devnet/pull/63/commits/325e3f7691ee59dc0c78415b1d4365dda1107ea3

**Comment by jcstein** [comment]
i was able to build this locally with no issues, but hit error on `Error: unknown flag: --gateway.deprecated-endpoints`

```logs
2023-10-12T14:36:23.586Z        INFO    node    nodebuilder/init.go:72  Node Store initialized
WARNING: Keep this auth token secret **DO NOT** log this auth token outside of development. CELESTIA_NODE_AUTH_TOKEN=

WARNING: Celestia custom network specified. Only use this option if the node is freshly created and initialized.
**DO NOT** run a custom network over an already-existing node store!

eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJwdWJsaWMiLCJyZWFkIiwid3JpdGUiLCJhZG1pbiJdfQ.bRQdLDQNuk5M1qWL-9HaAATVerU1Vsqklu8ZI09Cdlc
Error: unknown flag: --gateway.deprecated-endpoints
```

**Comment by jcstein** [comment]
testing with @smuu now, if we use GHCR it's compatible with AMD, but if we use dockerhub it is compatible with ARM

**Comment by nashqueue** [comment]
Can be closed in favor of #67 Thank you for debugging @jcstein @Bidon15 @smuu @MSevey

**Pull request closed without merging**